File: csharp/perfect-numbers/PerfectNumbers.cs
using System.Linq;

class PerfectNumbers
{
	public enum NumberType
	{
		Deficient, Abundant, Perfect
	}

	public static NumberType Classify(int number)
	{
		int sum = Enumerable.Range(1, number / 2).Where(x => number % x == 0).Sum();

		if (sum < number) return NumberType.Deficient;
		else if (sum > number) return NumberType.Abundant;
		else return NumberType.Perfect;
	}
}

File: javascript/perfect-numbers/perfect-numbers.js
export const classify = (n) => {
  if (n < 1)
    throw new Error('Classification is only possible for natural numbers.');
  else if (n === 1)
    return 'deficient';
  else {
    let sum = [...Array(Math.floor(n/2)).keys()]
      .map((_, i, ary) => ary[i] += 1)
      .filter(x => n % x === 0)
      .reduce((a, b) => a + b);

    if (sum === n) return 'perfect';
    else if (sum > n) return 'abundant';
    else return 'deficient';
  }
};


